'Solution'

KirillMelanich opened the pull request.

Grayder0152 commented on app/main.py:
```diff
@@ -1 +1,9 @@
-# write your code here
+def copy_file(command: str) -> None:
+    result = command.split()
```
check that command len == 2

Grayder0152 commented on app/main.py:
```diff
@@ -1 +1,9 @@
+        with open(result[1], "r") as fin, open(result[2], "w+") as fout:
+            fout.write(fin.read())
+
+    copy_file("cp file.txt file2.txt")
```
why do you run this?

KirillMelanich commented:
i will change my code radikaly

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Никаких
вирусов.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

чт, 9 мар. 2023 г. в 17:59, Sergey ***@***.***>:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In app/main.py
> <https://github.com/mate-academy/py-copy-file/pull/363#discussion_r1131245287>
> :
>
> > @@ -1 +1,9 @@
> -# write your code here
> +def copy_file(command: str) -> None:
> +    result = command.split()
>
> check that command len == 2
> ------------------------------
>
> In app/main.py
> <https://github.com/mate-academy/py-copy-file/pull/363#discussion_r1131245837>
> :
>
> > @@ -1 +1,9 @@
> -# write your code here
> +def copy_file(command: str) -> None:
> +    result = command.split()
> +    if result[1] == result[2]:
> +        print("Same file")
> +    else:
> +        with open(result[1], "r") as fin, open(result[2], "w+") as fout:
> +            fout.write(fin.read())
> +
> +    copy_file("cp file.txt file2.txt")
>
> why do you run this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mate-academy/py-copy-file/pull/363#pullrequestreview-1333166005>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5QWUSHUMOKBGLRWNHM5SQ3W3H45XANCNFSM6AAAAAAVVJOFJ4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

ArsenPidhoretskyi commented on app/main.py:
```diff
@@ -1 +1,6 @@
-# write your code here
+def copy_file(command: str) -> None:
+    file1, file2 = command.lstrip("cp").split()
```
But if command not equal to cp your code will broke example:
mv file1.txt file2.txt

ArsenPidhoretskyi replied:
file1 and file2 is not good names

ArsenPidhoretskyi reviewed (changes requested):
https://github.com/mate-academy/py-copy-file#note-check-your-code-using-this-checklist-before-pushing-your-solution

Grayder0152 commented on app/main.py:
```diff
@@ -1 +1,7 @@
+    if len(command.strip()) == 3:
+        prefix, source_file_path, destination_file_path = command.strip()
+        if source_file_path != destination_file_path and prefix == "cp":
+            with open(source_file_path, "r") as source_file, \
```
use `()` instead of `\`
